Shared/Samples/Find route in mobile map package/FindRouteInMobileMapPackageView.Model.swift

```diff
@@ -0,0 +1,195 @@
+        let routeTask: RouteTask?
+        
+        /// The route parameters for routing with the route task.
+        var routeParameters: RouteParameters?
+        
+        /// The locator task for reverse geocoding.
+        private let locatorTask: LocatorTask
+        
+        /// The parameters for reverse geocoding with the locator task.
+        private let reverseGeocodeParameters = {
+            let reverseGeocodeParameters = ReverseGeocodeParameters()
+            reverseGeocodeParameters.addResultAttributeName("*")
+            reverseGeocodeParameters.maxResults = 1
+            return reverseGeocodeParameters
+        }()
+        
+        /// The count of marker graphics in the marker graphics overlay.
+        private var markersCount: Int {
+            markerGraphicsOverlay.graphics.count
+        }
+        
+        /// The last marker in the marker graphics overlay.
+        var lastMarker: Graphic? {
+            markerGraphicsOverlay.graphics.last
+        }
+        
+        init(map: Map, locatorTask: LocatorTask) {
+            self.map = map
+            self.locatorTask = locatorTask
+            
+            // Set up the properties used for routing if the map has a transportation network.
+            if let transportationNetwork = map.transportationNetworks.first {
+                routeGraphicsOverlay = GraphicsOverlay()
+                routeTask = RouteTask(dataset: transportationNetwork)
+            } else {
+                routeGraphicsOverlay = nil
+                routeTask = nil
+            }
+        }
+        
+        // MARK: Methods
+        
+        /// Updates the marker to a given point or adds a new marker if there isn't one yet.
+        /// - Parameter point: The point to set the marker to.
+        func updateMarker(to point: Point) {
+            if lastMarker != nil {
+                lastMarker?.geometry = point
+            } else {
+                let markerGraphic = Graphic(geometry: point, symbol: markerSymbol)
+                markerGraphicsOverlay.addGraphic(markerGraphic)
+            }
+        }
+        
+        /// Adds a stop to the route.
+        /// - Parameters:
+        ///   - point: The point to add the stop graphic at.
+        func addRouteStop(at point: Point) async throws {
+            // Update the last stop instead of adding a new one if there
+            // isn't a route for the last one.
+            if routeGraphicsOverlay?.graphics.count ?? 0 < markersCount - 1 {
+                lastMarker?.geometry = point
+            } else {
+                addStopGraphic(at: point)
+            }
+            
+            // Create a route using the stop.
+            try await addRoute()
+        }
+        
+        /// Reverse geocodes a given point.
+        /// - Parameter point: The point to reverse geocode.
+        /// - Returns: The resulting address of the reverse geocode if any.
+        func reverseGeocode(point: Point) async throws -> String {
```

Comment:
Typically it is a good practice to handle the localization of strings in the view. But since geocoding results are localized in the back-end, where the server returns an already localized set of attributes based on request parameters, I think it makes sense to concat the strings in the model.